shot-android/src/main/java/com/karumi/shot/compose/ScreenshotSaver.kt

```diff
@@ -0,0 +1,53 @@
+package com.karumi.shot.compose
+
+import android.annotation.SuppressLint
+import android.graphics.Bitmap
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.ui.test.SemanticsNodeInteraction
+import java.io.File
+import java.io.FileOutputStream
+
+class ScreenshotSaver(private val packageName: String, private val bitmapGenerator: SemanticsNodeBitmapGenerator) {
+
+    @SuppressLint("SdCardPath")
+    private val screenshotsFolder: String = "/sdcard/screenshots/$packageName/screenshots-default/"
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun saveScreenshot(screenshot: ScreenshotToSave) {
+        val bitmap = bitmapGenerator.generateBitmap(screenshot)
+        createScreenshotsFolderIfDoesNotExist()
+        saveScreenshotBitmap(bitmap, screenshot.data)
+        bitmap.recycle()
+    }
+
+    private fun createScreenshotsFolderIfDoesNotExist() {
+        val folder = File(screenshotsFolder)
+        if (!folder.exists()) {
+            folder.mkdirs()
+        }
+    }
+
+    fun saveMetadata(session: ScreenshotTestSession) {
+        // TODO: Persist json metadata here with all the session information
+    }
+
+    private fun saveScreenshotBitmap(bitmap: Bitmap, data: ScreenshotMetadata) {
+        val screenshotPath = getScreenshotSdCardPath(data)
+        deletePreviousScreenshotIfExists(screenshotPath)
+        FileOutputStream(screenshotPath).use { out ->
+            bitmap.compress(Bitmap.CompressFormat.PNG, 100, out)
```

Comment:
are we using any kind of standard about using name lambda values instead 'it' ? Both are good for me, it's only a question for learning.

Reply:
As far as I know, there is no convention for now.